_posts/2018/2018-05-27-first-and-last-may-not-mean-what-you-think.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Active Record’s first and last may not mean what you think
+description: 'Naming truly is an art'
+layout: article
+category: ruby
+image:
+  base: '2018/first-and-last-may-not-mean-what-you-think'
+  alt: 'Running track'
+  source: 'https://unsplash.com/photos/52p1K0d0euM'
+  credit: 'Austris Augusts'
+---
+
+Each Active Record model comes with the well-used ‘get me a single record’ scopes: `.first` and `.last`.
+
+## Instead of…
+
+...using the methods directly on the base ActiveRecord class.
+
+```ruby
+User.first
+User.last
+```
+
+
+## Use…
+
+...them only with ordered scopes. Ideally with named concepts.
+
+### `app/models/user.rb`
+
+```ruby
+class User < ActiveRecord::Base
+  scope :by_created, -> { order(created_at: :asc) }
+  scope :earliest_created, -> { by_created.first }
+  scope :most_recently_created, -> { by_created.last }
+  # ...
+end
+
+User.most_recently_created
+```
+
+
+## But why?
+
+This is mostly about making the code more explicit.
+
+The `.first` and `.last` methods, when called on a bare Active Record model scope, are colloquially used to mean ‘created first’ and ‘created most recently’.
+
+It’s only an accident of Rails default design that sorting by ascending `id` is the same order as the ascending `created_at` timestamp.
+
+Using these scopes with no explicit order means you are relying heavily on the Rails convention of an incrementing integer `id`.
+
+You might remember I advocated [using UUIDs for primary keys](/2017/choose-uuids-for-model-ids-in-rails) in a previous article. This pairs up nicely don’t you think?
```

Comment:
With your "This pairs up nicely don't you think?" are you saying: if using UUIDs, this throws this whole 'first' and 'last' thing to shit?

Maybe get rid of that last question, and depending on how the preceding paragraph ends change it to something like: For example, imagine if you took my advice from a previous article where I advocated using UUIDs for primary keys?

Just a suggestion — tricky to see how it'll fit together without previously suggested edits.